pkg/dbmate/db.go

```diff
@@ -85,28 +85,44 @@ func (db *DB) Wait() error {
 	return fmt.Errorf("unable to connect to database: %s", err)
 }
 
-// CreateAndMigrate creates the database (if necessary) and runs migrations
-func (db *DB) CreateAndMigrate() error {
+// CreateAndMigrate creates the database (if necessary) and runs migrations.
+// If withRollback is true, this doesn't create or migrate but
+// returns if creation of database and applying of migrations would succeed.
+func (db *DB) CreateAndMigrate(withRollback bool) error {
 	drv, err := db.GetDriver()
 	if err != nil {
 		return err
 	}
 
+	if !drv.SupportsTransactionalDDL() && withRollback {
+		return fmt.Errorf("%s doesn't support Transactional DDL. "+
```

Comment:
No need for string concat here, just make it one long string

Reply:
golint doesn't seem to like lines longer than 100 columns.

Reply:
Lemme know if there is someway around golint rules